icechunk/src/format/transaction_log.rs

```diff
@@ -0,0 +1,73 @@
+use std::collections::{HashMap, HashSet};
+
+use serde::{Deserialize, Serialize};
+
+use crate::change_set::ChangeSet;
+
+use super::{
+    format_constants,
+    snapshot::{NodeSnapshot, NodeType},
+    ChunkIndices, IcechunkFormatVersion, NodeId,
+};
+
+#[derive(Clone, Debug, PartialEq, Default, Serialize, Deserialize)]
+pub struct TransactionLog {
+    // FIXME: better, more stable on-disk format
+    pub icechunk_transaction_log_format_version: IcechunkFormatVersion,
+    pub new_groups: HashSet<NodeId>,
```

Comment:
So i understand correctly, we keep `NodeId` for teh changes, which the `Repository` can then resolve into a more readbale format by resolving the zarr path from the ID in the given manifest right? Does the NodeId refer to the nodes id in the incoming changeset or from the existing changset?

Reply:
Yes, that is correct. NodeIds are slightly better than paths, because they are random, so something like `create /foo -> delete /foo -> create /foo` doesn't get tricky.

This is the NodeId assigned in the change that generate this transaction, so in your description, the existing changeset.